LEAF_Request_Portal/sources/Form.php

```diff
@@ -335,140 +335,148 @@ public function getForm($recordID, $limitCategory = null)
-        {
+
+        if (count($categoryIDsArr) === 0) {
             return 'Error: No forms selected. Please Select a form and try again.';
         }
 
         //check category visible status.  unpublished forms (visibility -1) should not be created.
-        $pubVars = array(
-            ":categoryIDs" => implode(',', $categoryIDsArr)
-        );
-        $pubSQL = "SELECT * FROM categories WHERE FIND_IN_SET(categoryID, :categoryIDs) AND `visible` = -1";
-        $unpublishedForms = $this->db->prepared_query($pubSQL, $pubVars);
+        $placeholders = implode(',', array_fill(0, count($categoryIDsArr), '?'));
+
+
+        $sql = "SELECT *
+                FROM `categories`
+                WHERE `categoryID` IN ({$placeholders})
+                AND `visible` = -1";
+
+        $unpublishedForms = $this->db->prepared_query($sql, $categoryIDsArr);
 
         if (count($unpublishedForms) > 0) {
             return 'Forms associated with this request are unpublished.  Please check the New Request page or contact an administrator about possible form updates.';
         }
 
-        $var = array(':service' => $_POST['service']);
+        $var = [':service' => $_POST['service']];
         $res = null;
-        if (is_numeric($_POST['service']))
-        {
-            $res = $this->db->prepared_query('SELECT serviceID FROM services WHERE serviceID=:service', $var);
-        }
-        else
-        {
-            $res = $this->db->prepared_query('SELECT serviceID FROM services WHERE service=:service', $var);
+
+        if (is_numeric($_POST['service'])) {
+            $sql = 'SELECT `serviceID`
+                    FROM `services`
+                    WHERE `serviceID` = :service';
+        } else {
+            $sql = 'SELECT `serviceID`
+                    FROM `services`
+                    WHERE `service` = :service';
         }
 
+        $res = $this->db->prepared_query($sql, $var);
+
         $serviceID = $res[0]['serviceID'];
-        if (!is_numeric($serviceID))
-        {
-            if ($_POST['service'] == 0)
-            {
+
+        if (!is_numeric($serviceID)) {
+            if ($_POST['service'] == 0) {
                 $serviceID = 0;
-            }
-            else
-            {
+            } else {
                 return 'Error: Service ID is not synchronized to Org. Chart.';
             }
         }
 
         $userMetadata  = $this->employee->getInfoForUserMetadata($userID, false);
-        $vars = array(
-            ':date' => time(),
-            ':serviceID' => $serviceID,
-            ':userID' => $userID,
-            ':title' => XSSHelpers::sanitizer($_POST['title']),
-            ':priority' => $_POST['priority'],
-            ':userMetadata' => $userMetadata,
-        );
+        $vars = [':date' => time(),
+                ':serviceID' => $serviceID,
+                ':userID' => $userID,
+                ':title' => XSSHelpers::sanitizer($_POST['title']),
+                ':priority' => $_POST['priority'],
```

Comment:
Does $_POST['priority'] get checked/sanitized at some point?  I can imagine Fortify having a hissy with that.